src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,100 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (newMovie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [movie, setMovie] = useState<Movie>({
+    title: '',
+    description: '',
+    imdbUrl: '',
+    imgUrl: '',
+    imdbId: '',
+  });
+
+  const isValidUrl = (url: string) => {
+    const pattern =
+      // eslint-disable-next-line
+      /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
+    return pattern.test(url);
+  };
+
+  const isFormValid =
+    movie.title.trim() &&
+    movie.imdbId.trim() &&
+    isValidUrl(movie.imgUrl.trim()) &&
+    isValidUrl(movie.imdbUrl.trim());
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (isFormValid) {
+      onAdd(movie);
+
+      setMovie({
+        title: '',
+        description: '',
+        imdbUrl: '',
+        imgUrl: '',
+        imdbId: '',
+      });
+    }
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form className="NewMovie" onSubmit={handleSubmit}>
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={movie.title}
+        onChange={value => setMovie({ ...movie, title: value })}
```

Comment:
```suggestion
        onChange={onChange}
```
Consider creating one handler for all fields